pytorch/training/docker/2.0/py3/cu121/Dockerfile.gpu

```diff
@@ -319,3 +318,214 @@ RUN rm -rf /root/.cache | true
+  -c nvidia \
+  -c conda-forge \
+  && /opt/conda/bin/mamba clean -afy
+
+# Install NCCL
+RUN cd /tmp \
+ && git clone https://github.com/NVIDIA/nccl.git -b v${NCCL_VERSION}-1 \
+ && cd nccl \
+ && make -j64 src.build BUILDDIR=/usr/local CUDA_LIB=${CUDA_HOME}/lib \
+ && rm -rf /tmp/nccl
+
+# install PyTorch related python packages (depends on PyTorch)
+RUN /opt/conda/bin/mamba install -y -c conda-forge \
+  # needed by fastai.distributed
+  accelerate \
+  && /opt/conda/bin/mamba install -y -c fastai fastai \
+  # fastai conda strictly depends on dataclasses
+  && pip uninstall -y dataclasses \
+  && pip install torchtnt \
+# explicitly install triton build-time dependencies, which won't get installed by conda for some reason
+  cmake \
+  lit \
+  && /opt/conda/bin/mamba clean -afy
+
+# Patches
+RUN pip install "pillow>=9.5" opencv-python huggingface_hub
+RUN /opt/conda/bin/mamba install -y -c conda-forge \
+  "requests>=2.31.0" \
+  && /opt/conda/bin/mamba clean -afy
+
+# Install GDRCopy which is a dependency of SM Distributed DataParallel binary
+# The test binaries requires cuda driver library which could be found in conda
+# So update the linker path to point to it to avoid -Lcuda not found
+RUN cd /tmp \
+ && git clone https://github.com/NVIDIA/gdrcopy.git -b v${GDRCOPY_VERSION} \
+ && cd gdrcopy \
+ && sed -ie '12s@$@ -L $(CUDA)/lib/stubs@' tests/Makefile \
+ && CUDA=${CUDA_HOME} make install \
+ && rm -rf /tmp/gdrcopy
+
+# Install Nvidia Apex (needs pytorch)
+## Pin apex commit requested by sm-model-parallel team
+RUN pip install packaging \
+  && git clone https://github.com/NVIDIA/apex \
+  && cd apex \
+  # requested by Rubik. Will wait for resolution on the illegal mem access error
+  && git checkout aa756ce \
+  && sed -i 's/from torch._six import string_classes//g' apex/amp/_initialize.py \
+  && sed -i 's/string_classes/str/g' apex/amp/_initialize.py \
+  && pip install -v --no-cache-dir --global-option="--cpp_ext" --global-option="--cuda_ext" ./ \
+  && cd .. \
+  && rm -rf apex
+
+# Install libboost from source. This package is needed for smdataparallel functionality [for networking asynchronous IO].
+RUN wget https://sourceforge.net/projects/boost/files/boost/1.73.0/boost_1_73_0.tar.gz/download -O boost_1_73_0.tar.gz \
+ && tar -xzf boost_1_73_0.tar.gz \
+ && cd boost_1_73_0 \
+ && ./bootstrap.sh \
+ && ./b2 threading=multi --prefix=/opt/conda -j 64 cxxflags=-fPIC cflags=-fPIC install || true \
+ && cd .. \
+ && rm -rf boost_1_73_0.tar.gz \
+ && rm -rf boost_1_73_0 \
+ && cd /opt/conda/include/boost
+
+WORKDIR /opt/pytorch
+
+# Copy workaround script for incorrect hostname
+COPY changehostname.c /
+COPY start_with_right_hostname.sh /usr/local/bin/start_with_right_hostname.sh
+
+RUN chmod +x /usr/local/bin/start_with_right_hostname.sh
+
+WORKDIR /root
+
+RUN pip install --no-cache-dir -U \
+    smclarify \
+    "sagemaker>=2,<3" \
+    "sagemaker-experiments<1" \
+    sagemaker-pytorch-training \
+    triton==2.0.0.dev20221202
```

Comment:
I guess we inherited this from 2.0 release? any context on pinged version?

Reply:
yes its inherited, I can unpin it as there are no SM features

Reply:
there should be no SM features needs a pinged version of triton.